behave/features/steps/specialist_provider.py

```diff
@@ -366,5 +370,67 @@ def step_impl(context):
     assert CASE_SPLIT_TEXT in full_case_log, "Could not find split case in operators dashboard"
 
 
+@step(u'I am on the CSV upload page')
+def step_impl(context):
+    csv_page = f"{CLA_FRONTEND_URL}/provider/csvupload/"
+    context.helperfunc.open(csv_page)
+    page = f"/provider/csvupload/"
+    wait_until_page_is_loaded(page, context)
+    header = context.helperfunc.find_by_xpath("//header/h2[text()='CSV Upload']")
+    assert header is not None
+
+
+@step(u'I select \'Choose file\' and upload a csv file')
+def step_impl(context):
+    input_csv = context.helperfunc.find_by_xpath("//input[@name='csvfile']")
+    # test will fail if file can't be found or upload fails.
+    input_csv.send_keys(os.getcwd() + "/uploads/csvupload.csv")
+
 
+@step(u'I select the month and year for the uploaded csv file')
+def step_impl(context):
+    date_now = datetime.datetime.now().date()
+    # Alter date to be the first day of month
+    # Minis one month because current month will not be visible as option
+    new_date = date_now.replace(day=1, month=date_now.month - 1).strftime("%Y-%m-%d")
+    select = Select(context.helperfunc.find_by_xpath(f'//form[@name="csvForm"]/label/select'))
+    assert select is not None
+    try:
+        select.select_by_value(f"string:{new_date}")
+    except StaleElementReferenceException:
+        assert False, f"Could find date month or year in select options"
+
+
+@step(u'I select the \'upload\' button and check for errors')
+def step_impl(context):
+    # perfect for buttons with submit types
+    button = context.helperfunc.find_by_xpath("//button[@type='submit'][text()='Upload']")
+    assert button is not None
+    button.click()
 
+    # check there are no duplicates or problems with the CSV file
+    error_header = "//h2[text()='Please correct the following errors:']"
+    error_notice = "//div[1]/ul/li"
+
+    context.csv_page = context.helperfunc.find_by_xpath("//*[@id='wrapper']")
```

Comment:
Can you combine these checks for error messages? Maybe one try catch with an `if context.csv_page.find_element_by_xpath(error_notice) OR context.csv_page.find_element_by_xpath(error_header)`? 
I am not sure it matters where the error message is, maybe just print out what the error message is as that would help with any problems.

Reply:
Changed and added comments about what error_header, error_notice is.